solr/core/src/java/org/apache/solr/search/SortSpec.java

```diff
@@ -62,18 +62,15 @@ public void setSortAndFields(Sort s, List<SchemaField> fields) {
     this.fields = Collections.unmodifiableList(fields);
   }
 
-  public static boolean includesScore(Sort sort) {
+  public boolean includesScore() {
+    if (num == 0) return false;
```

Comment:
I observe `num` defaults to 10; some constructors here don't take this thus it keeps this default.  This is disconcerting; ideally the arg is mandatory; right?  If fixing that isn't necessary to accomplish the performance goal then okay to defer.